packages/puppeteer-crawler/src/internals/utils/puppeteer_utils.ts

```diff
@@ -603,6 +604,41 @@ export async function saveSnapshot(page: Page, options: SaveSnapshotOptions = {}
     }
 }
 
+export interface PuppeteerContextUtils {
+    injectFile(filePath: string, options?: InjectFileOptions): Promise<unknown>;
+    injectJQuery(): Promise<unknown>;
+    enqueueLinksByClickingElements(options: Omit<EnqueueLinksByClickingElementsOptions, 'page' | 'requestQueue'>): Promise<storage.BatchAddRequestsResult>;
+    blockRequests(options?: BlockRequestsOptions): Promise<void>;
+    blockResources(resourceTypes?: string[]): Promise<void>;
+    cacheResponses(cache: Dictionary<Partial<ResponseForRequest>>, responseUrlRules: (string | RegExp)[]): Promise<void>;
+    compileScript(scriptString: string, ctx?: Dictionary): CompiledScriptFunction;
+    addInterceptRequestHandler(handler: InterceptHandler): Promise<void>;
+    removeInterceptRequestHandler(handler: InterceptHandler): Promise<void>;
+    infiniteScroll(options?: InfiniteScrollOptions): Promise<void>;
+    saveSnapshot(options?: SaveSnapshotOptions): Promise<void>;
+}
+
+/** @internal */
+export function registerUtilsToContext(context: PuppeteerCrawlingContext): void {
+    context.injectFile = (filePath: string, options?: InjectFileOptions) => injectFile(context.page, filePath, options);
+    context.injectJQuery = () => injectJQuery(context.page);
+    context.enqueueLinksByClickingElements = (options: Omit<EnqueueLinksByClickingElementsOptions, 'page' | 'requestQueue'>) => enqueueLinksByClickingElements({
+        page: context.page,
+        requestQueue: context.crawler.requestQueue!,
+        ...options,
+    });
+    context.blockRequests = (options?: BlockRequestsOptions) => blockRequests(context.page, options);
+    context.blockResources = (resourceTypes?: string[]) => blockResources(context.page, resourceTypes);
+    context.cacheResponses = (cache: Dictionary<Partial<ResponseForRequest>>, responseUrlRules: (string | RegExp)[]) => {
+        return cacheResponses(context.page, cache, responseUrlRules);
+    };
+    context.compileScript = (scriptString: string, ctx?: Dictionary) => compileScript(scriptString, ctx);
```

Comment:
I've put all of them here, `compileScript` seems to be the only one that does not benefit from being bound to context, but having a single utility outside of context feels weird

only exception I haven't put here is `gotoExtended`, do we want that here too? it was used directly in the crawler, not sure if it makes sense to expose it for users this way?